smtp_connector.py

```diff
@@ -800,6 +817,12 @@ def _send_email(self, param, action_result):
         sender_address = config.get('sender_address', config.get(phantom.APP_JSON_USERNAME))
         email_from = param.get(SMTP_JSON_FROM, sender_address)
 
+        # validate sender email if inputted as a parameter
+        if action_id != "test_connectivity" and param.get(SMTP_JSON_FROM):
```

Comment:
If you do it like this, then it looks like someone could override this check by using `config.get(phantom.APP_JSON_USERNAME)` instead of `param.get(SMTP_JSON_FROM)`. Consider validating on the value that is actually used further down in the code. Same in the other place this is done

Reply:
Hello @bb-splunk,
Following @leobao-splunk  guidance, we have adjusted our approach to email address validation for Splunk's internal SMTP assets. Since these assets might contain restricted domains, validating them upfront would cause compatibility issues.  Therefore, email validation will only occur for email addresses provided during action runs. This ensures accurate validation without interfering with the functionality of internal assets.

Reply:
I'm sorry but I don't understand what you're trying to say. Adding extra validations can introduce issues anyways, and yes, test connectivity is excluded, but how does that address my point?

Reply:
As discussed with @leobao-splunk, we don't want to perform validation whenever the input for the `from` field is taken from the asset configuration 
https://splunk.slack.com/archives/D072E46B5L3/p1715354173342229